lib/ts/recipe/emailverification/utils.ts

```diff
@@ -44,10 +37,41 @@ export function validateAndNormaliseUserInput(
         ...config.override,
     };
 
+    let emailService = config.emailDelivery?.service;
+    /**
+     * following code is for backward compatibility.
+     * if user has not passed emailDelivery config, we
+     * use the createAndSendCustomEmail config. If the user
+     * has not passed even that config, we use the default
+     * createAndSendCustomEmail implementation which calls our supertokens API
+     */
+    if (emailService === undefined) {
+        emailService = new BackwardCompatibilityService(
+            appInfo,
+            recipe.isInServerlessEnv,
```

Comment:
This value may still be `undefined` by the time it comes here (if we call `validateAndNormaliseUserInput` before setting `isInServerlessEnv` in the recipe). So instead, pass the boolean for `isInServerlessEnv` to this function.